src/parser/jobs/brd/modules/OGCDDowntime.js

```diff
@@ -18,4 +19,37 @@ export default class OGCDDowntime extends CooldownDowntime {
 		this.target = 100
 	}
 
+	// This is to handle cases where the player casts Raging Strikes pre-pull
+	normalise(events) {
+		let firstRagingStrikesCast = 0
+		let firstRagingStrikesApply = 0
+
+		// Used to capture a Raging Strike cast to fabricate a new one easily
+		let ragingStrikesCast = {}
+
+		for (const event of events) {
+			// Checks to make sure it's the players cast of Raging Strikes
+			if (this.parser.byPlayer(event) && event.ability) {
+				if (!firstRagingStrikesCast && event.type === 'cast' && event.ability.guid === ACTIONS.RAGING_STRIKES.id) {
+					firstRagingStrikesCast = event.timestamp
+					ragingStrikesCast = event
+				}
+				if (!firstRagingStrikesApply && event.type === 'applybuff' && event.ability.guid === STATUSES.RAGING_STRIKES.id) {
+					firstRagingStrikesApply = event.timestamp
+				}
+				// Once we have gotten this information, no need to waste time
+				if (firstRagingStrikesCast && firstRagingStrikesApply) {
+					break
+				}
+			}
+		}
+
+		if (firstRagingStrikesCast > firstRagingStrikesApply) {
+			this.parser.fabricateEvent({
```

Comment:
`fabricateEvent` only works "properly" within the standard parser lifecycle. in a normaliser, you'll need to `splice` the event into the array